### PR TITLE
Feature/users

### PR DESCRIPTION
**Pull request by michaeljcollinsuk**
Initial implementation of front end login functionality

- Custom user model to allow login via email
- Using `django-sesame` to build magic link urls rather than password authentication

### CODE REVIEW

**Comment by michaeljcollinsuk** [comment]
The `pipenv check` error is strange as the dependency causing the error is not in the Pipfile.lock https://app.circleci.com/pipelines/github/DemocracyClub/ec-api-proxy/49/workflows/1cd9a964-719a-4854-84ef-c7903c88698e/jobs/140/parallel-runs/0/steps/0-105?invite=true

File: ec_api/apps/users/forms.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        fields = ["email"]
+        error_messages = {
+            "email": {
+                "unique": "User with this email has already registered. Do you need to log in?"
+            }
+        }
+
+    def save(self, commit=True):
+        """
+        Password field is not used to set unusable password before saving
```
Comment (GeoWill) [inline ec_api/apps/users/forms.py:25]: Possible small typo: should 'Password field is not used to set' be 'Password field is not used, so set'? 
If not, I don't understand the comment (but that will be due to me not knowing about handling passwords in django... :grinning: )

Comment (michaeljcollinsuk) [reply ec_api/apps/users/forms.py:25]: Yep... that sentence makes no sense at all 😆! I'll update to something that does, thanks

File: ec_api/apps/users/forms.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    """
+    Registration form for a User
+    """
+
+    class Meta:
+        model = User
+        fields = ["email"]
+        error_messages = {
+            "email": {
+                "unique": "User with this email has already registered. Do you need to log in?"
```
Comment (symroe) [inline ec_api/apps/users/forms.py:19]: Is this ever used?

Comment (michaeljcollinsuk) [reply ec_api/apps/users/forms.py:19]: It is if you try to register with an email that is already registered... 
Screenshot 2021-03-02 at 10 49 53

Whether or not we want to show it is another question though. I know we don't for login (will always just display "check your email") but wasn't sure what's best practice for registration

Comment (symroe) [reply ec_api/apps/users/forms.py:19]: But in this situation we can just send the token? As in, we don't need to have different forms for each action, when they're always the same?

**Pull request merged**